test/src/actions/TestGetActionsAction.C

```diff
@@ -0,0 +1,75 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "TestGetActionsAction.h"
+
+#include "ActionWarehouse.h"
+#include "AddMaterialAction.h"
+#include "MooseApp.h"
+
+#include <fstream>
+
+template<>
+InputParameters validParams<TestGetActionsAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("An action demonstrating how an action can interact with other actions");
+  params.addParam<FileName>("csv_file", "test_get_actions.csv", "CSV file for printing the action information");
+  params.addParam<processor_id_type>("pid", 0, "Test the output on a designated processor, default to master");
+
+  // A switch for changing the function that is being tested
+  MooseEnum test_function("getAction getActions");
+  params.addRequiredParam<MooseEnum>("test_function", test_function, "The function to test: getActions or getAction.");
+
+  return params;
+}
+
+TestGetActionsAction::TestGetActionsAction(const InputParameters & params) :
+    Action(params),
+    _test_function(getParam<MooseEnum>("test_function"))
```

Comment:
While this is true you end up with a switch statement that looks like the one below:

```
  case 1: // FOO
  case 2: // BAR
```

If you create a scoped enum in the class it just looks cleaner and isn't any more expensive to run or execute. Underneath the hood they are still just integers (or some other simple POD type):
```
    case ClassEnum::Foo:
    case ClassEnum::Bar:
```